packages/@aws-cdk/core/lib/private/synthesis.ts

```diff
@@ -49,7 +54,66 @@ export function synthesize(root: IConstruct, options: SynthesisOptions = { }): c
   // stacks to add themselves to the synthesized cloud assembly.
   synthesizeTree(root, builder, options.validateOnSynthesis);
 
-  return builder.buildAssembly();
+  const assembly = builder.buildAssembly();
+
+  invokeValidationPlugins(root, builder.outdir);
+
+  return assembly;
+}
+
+function invokeValidationPlugins(root: IConstruct, outdir: string) {
+  const originalHash = computeChecksumOfFolder(outdir);
+  const tree = new ConstructTree(root);
```

Comment:
We should run this _only_ if there are validation plugins.

Reply:
I made this computation lazy.